TCD1304.py

```diff
@@ -54,36 +57,36 @@ def __init__(
         self.min_voltagerange_clock = min_voltagerange_clock
         self.max_voltagerange_clock = max_voltagerange_clock
         self.pwmgen = PWMGenerator() #once assigned in __init__ self.pwmgen can be used in other methods
+        self.scope = Oscilloscope ()
+        
         
     def power_source ():        # puts a voltage of 4V on PV1
         ps = PowerSupply()
         ps.pv1 = 4
         ps.pv1
     
     def master_clock (self):        # puts PWM with frequency of 2MHz on SQ1; masterclock is the fastest clock
-        prescaler = int(math.log(oscillator_frequency / frequency_master_clock) / math.log(2))   # When setting a frequency by mapping the reference clock directly to a PWM output, only frequencies which are even factors of 128 MHz (the frequency of the PSLab's main oscillator) are available. The frequency is therefore not set by specifying the frequency itself, but by setting a prescaler.
         self.pwmgen.map_reference_clock(["SQ1"], prescaler)    
     
-    def sh_clock (self):           #  on SQ2 sets pulse for the integrationtime
-        return self.pwmgen.generate() #is the return attribute the one to choose in order to set a second clock?
+    def sh_clock (self):           #  on SQ2 sets pulse for the integrationtime. Running the Sensor in shutter mode try with a frequency of 1/4 of master_clock. tint (min) = 10µs
+        self.pwmgen.generate(["SQ2"], prescaler/4 ) 
 
-        ''' 
-        integrationtime is started by the falling flank of voltageoutput. 
-        the time for integration must be longer than 3_694 Elements of fmc.
-        sh clock is started before icg is started. normal state of sh is a low
-        level. high level can be quite short
-        '''
 
     def icg_clock (self):           #  on SQ3 starts and stops the reading of the sensor; this third clock is the slowest clock
         self.pwmgen.set_state()
-            for pulse in range(integration_elements):
-                self.pwmgen.set_state(sq3=True)
-                time.sleep(integration_time / 2)
-                self.pwmgen.set_state(sq3=False)
-                time.sleep(integration_time / 2)
+         for pulse in range(integration_elements):
+            self.pwmgen.set_state(sq3=True)
+            time.sleep(integration_time / 2)
+            self.pwmgen.set_state(sq3=False)
+            time.sleep(integration_time / 2)
         
         '''
         normal state of icg is a high level. Start of the reading is on the rising
         flank of the voltage. To start the reading voltage shall drop to low and rise 
         to high again.
         '''
+    def analog_signal_read (self):
+        self.scope = Oscilloscope()
+        x, y = scope.capture (1, integration_elements, (integration_time / integration_elements)*microseconds_in_second) # Parameters set are (Ch1, amount of samples(3694), time between samples (~0,5µs)) 

```

Comment:
You will probably need to use the trigger functionality to start sampling at the right moment. When a trigger is active the time gap (`integration_time / integration_elements`) must be 1 µs or greater.

Reply:
Okay, cool. We try to implement it. ...Thank you very much for you constant input Alex. We value that a lot. Wouldn't be possible without you.